src/OrchardCore/Orchard.Environment.Shell/RunningShellTable.cs

```diff
@@ -100,24 +100,38 @@ public ShellSettings Match(string host, string appRelativePath)
         private bool TryMatchInternal(string host, string appRelativePath, out ShellSettings result)
         {
             // 1. Search for Host + Prefix match
-            string hostAndPrefix = GetHostAndPrefix(host, appRelativePath);
+            var hostAndPrefix = GetHostAndPrefix(host, appRelativePath);
 
             if (!_shellsByHostAndPrefix.TryGetValue(hostAndPrefix, out result))
             {
-                // 2. Search for Host only match
+                // 2. Search for Full Host only match
```

Comment:
Its something that came up where I am working... and why not? Its a valid use case.

They would rather use ports than virtual paths or host entries.